server/src/main/java/io/whitefox/api/deltasharing/server/DeltaSharesApiImpl.java

```diff
@@ -58,8 +76,23 @@ public CompletionStage<Response> getTableChanges(
   @Override
   public CompletionStage<Response> getTableMetadata(
       String share, String schema, String table, String startingTimestamp) {
-    Response res = Response.ok().build();
-    return CompletableFuture.completedFuture(res);
+    // TODO handle deltaSharingCapabilities Header
+    // TODO streaming response
+    // TODO handle responseFormat
+    return deltaSharesService
+        .getPTable(share, schema, table)
+        .thenCompose(o -> optionalToNotFoundAsync(o, pTable -> deltaSharedTableTableLoader
```

Comment:
Isn’t there a way to not nest 3 completableFuture? Something like flat map in scala?

Reply:
I was just looking for this, and I realized that thenCompose is equivalent to flatMap. https://stackoverflow.com/questions/38885262/what-is-completablefutures-equivalent-of-flatmap